app/controllers/application_controller.rb

```diff
@@ -49,9 +49,12 @@ def check_form
   end
 
   def check_draw_period
+    url = request.fullpath
     unless current_draw_period != nil then
       unless current_user.is_admin then
-        redirect_to coming_soon_path
+        unless url == "/" or url.start_with?("/dorm") #Let students see the map but nothing else
```

Comment:
Does rails have a way to check what resources we are requesting?  Because `url.start_with?("/dorm?")` seems fragile: what if we eventually create something `/dormant_rooms` that we only want available during the draw period?